CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs

```diff
@@ -0,0 +1,227 @@
+﻿using CharlieBackend.Business.Services.Interfaces;
+using CharlieBackend.Core;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.DTO.Schedule;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+
+using CharlieBackend.Core.Models.ResultModel;
+using CharlieBackend.Business.Services.ScheduleServiceFolder;
+
+namespace CharlieBackend.Business.Services
+{
+    public class ScheduleService : IScheduleService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public ScheduleService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<Result<EventOccurrenceDTO>> CreateScheduleAsync(CreateScheduleDto createScheduleRequest)
+        {
+            string error = await ValidateCreateScheduleRequestAsync(createScheduleRequest);
+
+            if (error != null)
+            {
+                return Result<EventOccurrenceDTO>.GetError(ErrorCode.ValidationError, error);
+            }
+
+            EventOccurrence result = new EventOccurrence
+            {
+                Pattern = createScheduleRequest.Pattern.Type,
+                StudentGroupId = createScheduleRequest.Context.GroupID,
+                EventStart = createScheduleRequest.Range.StartDate,
+                EventFinish = createScheduleRequest.Range.FinishDate.Value,
+                Storage = EventOccuranceStorageParser.GetPatternStorageValue(createScheduleRequest.Pattern)
+            };
+
+            _unitOfWork.EventOccurenceRepository.Add(result);
+
+            _unitOfWork.ScheduledEventRepository.AddRange(new ScheduledEventHandlerFactory().Get(createScheduleRequest.Pattern).GetEvents(result, createScheduleRequest.Context));
```

Comment:
Please insert ISheduledEventHandlerFactory to the constructor dependency

Reply:
Updated

Reply:
Please create an interface for ScheduledEventHandlerFactory and use DI to get implementation of ScheduledEventHandlerFactory. It will help us to mock data and simplify unit testing.